nav2_costmap_2d/plugins/costmap_filters/keepout_filter.cpp

```diff
@@ -126,6 +133,7 @@ void KeepoutFilter::maskCallback(
 
   // Making a new mask_costmap_
   mask_costmap_ = std::make_unique<Costmap2D>(*msg);
+  mask_frame_ = msg->header.frame_id;
```

Comment:
In the new constructor we needed for `Costmap2D` to process occ grids, I don't see where it sets the `global_frame`. If you do that, then you don't need to store the `mask_frame_` here since it will be in the `mask_costmap_`.

Reply:
I am not sure that this is possible in current design:
Currently `global_frame_` is stored by `LayeredCostmap`, but not stored by `Costmap2D`. As far as I understand, this field should be the same for all layers by design (but maybe I am wrong).

So, when new `Costmap2D` is being constructed from `OccupancyGrid`, there is no extra field where to store the frame inside. So I've used a new variable for it.

Reply:
Got it - that's fine.